--- configure.in	2011-11-27 02:16:43.352291700 -0600
+++ configure.in	2011-11-27 02:16:15.000000000 -0600
@@ -76,11 +76,6 @@ if test x$srcdir != x.; then
     INCLUDE="-Iinclude $INCLUDE"
 fi
 case "$host" in
-    *-*-cygwin*)
-        # We build SDL on cygwin without the UNIX emulation layer
-        BASE_CFLAGS="-I/usr/include/mingw -mno-cygwin"
-        BASE_LDFLAGS="-mno-cygwin"
-        ;;
     *)
         BASE_CFLAGS="-D_GNU_SOURCE=1"
         BASE_LDFLAGS=""
@@ -156,7 +151,7 @@ if test x$enable_libc = xyes; then
 
     dnl Checks for library functions.
     case "$host" in
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ;;
     *)
         AC_FUNC_ALLOCA
@@ -467,7 +462,7 @@ AC_HELP_STRING([--enable-esd], [support
             AC_ARG_ENABLE(esd-shared,
 AC_HELP_STRING([--enable-esd-shared], [dynamically load ESD audio support [[default=yes]]]),
                           , enable_esd_shared=yes)
-            esd_lib=[`find_lib "libesd.so.*" "$ESD_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            esd_lib="cygesd-0.dll"
 
             AC_DEFINE(SDL_AUDIO_DRIVER_ESD)
             SOURCES="$SOURCES $srcdir/src/audio/esd/*.c"
@@ -514,7 +509,7 @@ AC_HELP_STRING([--enable-pulseaudio], [u
             AC_ARG_ENABLE(pulseaudio-shared,
 AC_HELP_STRING([--enable-pulseaudio-shared], [dynamically load PulseAudio support [[default=yes]]]),
                           , enable_pulseaudio_shared=yes)
-            pulse_lib=[`find_lib "libpulse-simple.so.*" "$PULSE_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            pulse_lib="cygpulse-simple-0.dll"
 
             AC_DEFINE(SDL_AUDIO_DRIVER_PULSE)
             SOURCES="$SOURCES $srcdir/src/audio/pulse/*.c"
@@ -622,7 +617,7 @@ AC_HELP_STRING([--enable-nas], [support
             AC_ARG_ENABLE(nas-shared,
 AC_HELP_STRING([--enable-nas-shared], [dynamically load NAS audio support [[default=yes]]]),
                           , enable_nas_shared=yes)
-            nas_lib=[`find_lib "libaudio.so.*" "$NAS_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            nas_lib="cygaudio-2.dll"
 
             if test x$have_loadso != xyes && \
                test x$enable_nas_shared = xyes; then
@@ -729,7 +724,7 @@ AC_HELP_STRING([--enable-nasm], [use nas
 
         if test x"$NASMFLAGS" = x; then
             case $ARCH in
-              win32)
+              win32|cygwin)
                   NASMFLAGS="-f win32"
                   ;;
               openbsd)
@@ -964,6 +959,13 @@ AC_HELP_STRING([--enable-x11-shared], [d
                           , enable_x11_shared=maybe)
 
             case "$host" in
+                *-*-cygwin*)
+                    x11_symbols_private=yes
+                    x11_lib='cygX11-6.dll'
+                    x11ext_lib='cygXext-6.dll'
+                    xrender_lib='cygXrender-1.dll'
+                    xrandr_lib='cygXrandr-2.dll'
+                    ;;
                 *-*-darwin*) # Latest Mac OS X actually ships with Xrandr/Xrender libs...
                     x11_symbols_private=yes
                     x11_lib='/usr/X11R6/lib/libX11.6.dylib'
@@ -2452,7 +2454,7 @@ case "$host" in
             have_timers=yes
         fi
         ;;
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ARCH=win32
         if test "$build" != "$host"; then # cross-compiling
             # Default cross-compile location
@@ -2524,6 +2526,45 @@ case "$host" in
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
         SDL_LIBS="-lmingw32 -lSDLmain $SDL_LIBS -mwindows"
         ;;
+    *-*-cygwin*)
+        ARCH=cygwin
+        CheckDummyVideo
+        CheckDiskAudio
+        CheckDummyAudio
+        CheckDLOPEN
+        CheckNASM
+        CheckOSS
+        CheckARTSC
+        CheckESD
+        CheckNAS
+        CheckPulseAudio
+        CheckX11
+        CheckGGI
+        CheckAAlib
+        CheckCaca
+        CheckOpenGLX11
+        CheckPTHREAD
+        # Set up files for the joystick library
+        if test x$enable_joystick = xyes; then
+            AC_DEFINE(SDL_JOYSTICK_WINMM)
+            SOURCES="$SOURCES $srcdir/src/joystick/win32/*.c"
+            have_joystick=yes
+        fi
+        # Set up files for the cdrom library
+        if test x$enable_cdrom = xyes; then
+            AC_DEFINE(SDL_CDROM_WIN32)
+            SOURCES="$SOURCES $srcdir/src/cdrom/win32/*.c"
+            have_cdrom=yes
+        fi
+        # Set up files for the timer library
+        if test x$enable_timers = xyes; then
+            AC_DEFINE(SDL_TIMER_UNIX)
+            SOURCES="$SOURCES $srcdir/src/timer/unix/*.c"
+            have_timers=yes
+        fi
+        # Set up the system libraries we need
+        EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lwinmm"
+        ;;
     *-wince*)
         ARCH=win32
         CheckDummyVideo
--- include/SDL_platform.h	2011-11-27 02:16:43.355291900 -0600
+++ include/SDL_platform.h	2011-11-27 02:16:15.000000000 -0600
@@ -43,6 +43,10 @@
 #undef __BSDI__
 #define __BSDI__	1
 #endif
+#if defined(__CYGWIN__)
+#undef __CYGWIN__
+#define __CYGWIN__	1
+#endif
 #if defined(_arch_dreamcast)
 #undef __DREAMCAST__
 #define __DREAMCAST__	1
--- src/video/x11/SDL_x11gl.c	2011-11-27 02:16:43.357292000 -0600
+++ src/video/x11/SDL_x11gl.c	2011-11-27 02:16:15.000000000 -0600
@@ -35,6 +35,8 @@
 #define DEFAULT_OPENGL	"libGL.so.3"
 #elif defined(__OpenBSD__)
 #define DEFAULT_OPENGL	"libGL.so.4.0"
+#elif defined(__CYGWIN__)
+#define DEFAULT_OPENGL	"cygGL-1.dll"
 #else
 #define DEFAULT_OPENGL	"libGL.so.1"
 #endif
